packages/obonode/obojobo-chunks-figure/choose-image-modal.js

```diff
@@ -0,0 +1,145 @@
+		}
+
+		this.firstRef = React.createRef()
+	}
+
+	componentDidMount() {
+		this.fetchMedias()
+	}
+
+	handleFileChange(event) {
+		const file = event.target.files[0]
+		const formData = new window.FormData()
+		formData.append('userImage', file, file.name)
+		APIUtil.postMultiPart('/api/media/upload', formData).then(mediaData => {
+			this.props.onCloseChooseImageModal(mediaData.media_id)
+		})
+	}
+
+	// Fetch `count` number of image
+	fetchMedias() {
+		this.setState({
+			...this.state,
+			isFetching: true,
+			hasMore: false
+		})
+		APIUtil.get(`/api/media/many/?start=${this.state.start}&count=${this.state.count}`)
+			.then(res => res.json())
+			.then(result => {
+				if (result.status !== 'error') {
+					this.setState({
+						...this.state,
+						medias: [...this.state.medias, ...result.data],
+						isFetching: false,
+						hasMore: result.hasMore,
+						start: this.state.start + this.state.count
+					})
+				} else {
+					this.setState({ ...this.state, isFetching: false, hasMore: false })
+				}
+			})
+			.catch(() => {
+				this.setState({ ...this.state, isFetching: false, hasMore: false })
+			})
+	}
+
+	focusOnFirstElement() {
+		return this.firstRef.current.focus()
+	}
+
+	onHandleKeyPress(event, callback) {
+		if (event.key === 'Enter') {
+			callback()
+		}
+	}
+
+	render() {
+		return (
+			<SimpleDialog
+				cancelOk
+				title="Choose an image"
+				onConfirm={() => this.props.onCloseChooseImageModal(this.state.url)}
+				onCancel={() => this.props.onCloseChooseImageModal(null)}
+				focusOnFirstElement={this.focusOnFirstElement.bind(this)}
+			>
+				<div className="choose-image">
+					<div className="choose-image--image-controls">
+						<div className="choose-image--image-controls--upload">
+							<input
+								tabIndex={0}
+								type="file"
+								name="file"
+								id="file"
+								className="inputfile"
+								ref={this.firstRef}
+								onChange={event => this.handleFileChange(event)}
+							/>
+							<label htmlFor="file">Upload new image</label>
+						</div>
+
+						<div className="choose-image--image-controls--or">or</div>
+
+						<input
+							id="choose-image--image-controls--url"
+							type="text"
+							placeholder="Enter URL"
+							value={this.state.url}
+							onChange={e => this.setState({ url: e.target.value })}
+						/>
+					</div>
+					<div className="choose-image--divider" />
+					<small>Recently uploaded</small>
+					<div className="choose-image--image-gallary">
+						{this.state.medias.map(media => (
+							<img
+								tabIndex={0}
+								className="image-gallary--single-photo"
+								key={media.id}
+								id={media.id}
+								src={`/api/media/${media.id}/small`}
+								onClick={() => this.props.onCloseChooseImageModal(media.id)}
+								onKeyPress={event =>
+									this.onHandleKeyPress(event, () => this.props.onCloseChooseImageModal(media.id))
+								}
+							/>
+						))}
+
+						{this.state.isFetching ? <h5>Loading...</h5> : null}
```

Comment:
Screen readers read out the "Loading..." regardless of whether or not the images have already loaded.

Reply:
I added `aria-live="polite"`. User will be notified when the UI is updated

Reply:
Testing this I always ended up hearing "Loading...". I tried using hidden, aria-busy and role="alert" since I was curious but I also always ended up hearing 'Loading'. I'm not sure the solution to this one.

Reply:
@qwertynerd97 Just checking if you have any ideas?

Reply:
The problem is that the content of the dialog is loaded into the read buffer with Loading..., and it reads that out, even though the content area has changed (also i think ChromeVox may have some issues with aria-live?) .  For the aria-live to work, it needs to be on an element that is loaded with the DOM - moving it up to the div might work since it's loaded with the modal, but I'm not certian.